### PR TITLE
2.0

### PR DESCRIPTION
**Pull request by aensidhe**

### CODE REVIEW

File: src/msgpack.light.benchmark/Data/BeerConverter.cs [hunk 1]
```diff
@@ -1,70 +1,92 @@
...
+            _floatParser = context.GetRequiredParser<float>();
+        }
+
+        public int GetBufferSize(Beer value) => value == null
+            ? DataLengths.Nil
+            : _stringFormatter.GetBufferSize(value.Brand)
+            + _stringFormatter.GetBufferSize(value.Brewery)
+            + _listStringFormatter.GetBufferSize(value.Sort)
+            + _floatFormatter.GetBufferSize(value.Alcohol)
+            + 27;
```
Comment (roman-kozachenko) [inline src/msgpack.light.benchmark/Data/BeerConverter.cs:34]: What is 27?

File: src/msgpack.light.benchmark/Data/BeerConverterHardcore.cs [hunk 2]
```diff
@@ -1,78 +1,114 @@
...
 
-        private IMsgPackConverter<float> _floatConverter;
+        private readonly IMsgPackParser<string> _stringParser;
+        private readonly IMsgPackParser<List<string>> _listStringParser;
+        private readonly IMsgPackParser<float> _floatParser;
 
-        private MsgPackContext _context;
+        static BeerConverterHardCore()
+        {
+            Brand = new byte[6];
```
Comment (roman-kozachenko) [inline src/msgpack.light.benchmark/Data/BeerConverterHardcore.cs:24]: `6 = nameof(Beer.Brand).Length + 1` 
Why did you add `1`?

**Review by roman-kozachenko — COMMENTED** [review]
Do we really need to split Formatter and Parser?
I think more often you will implement both of them, and you have to resolve both Formatters and Parsers for all class properties. One of the solution is to introduce 
`IMsgPackConveter<T>: IMsgPackParser<T>, IMsgPackFormatter<T>` and `context.GetRequeiredConverter`.
Another point of possible simplification - generalize Usual/Constrainted formatters somehow. Also, usage of Constainted formatters is unclear

File: src/msgpack.light.benchmark/Data/BeerConverterHardcore.cs (hunk 2 above)
Comment (aensidhe) [reply src/msgpack.light.benchmark/Data/BeerConverterHardcore.cs:24]: `DataCodes.FixStr` header

File: src/msgpack.light.benchmark/Data/BeerConverter.cs (hunk 1 above)
Comment (aensidhe) [reply src/msgpack.light.benchmark/Data/BeerConverter.cs:34]: Magic number! :-) Headers and lengths of property names.
